### ISSUE TITLE
Metadata.hide affects fields in the submission process

### ISSUE THREAD

**Issue by S-Eiberger** [DSpace/DSpace#9092]
**Describe the bug**
In the submission process, the value of metadata fields that are listed in metadata.hide are only displayed to administrators after saving. For all other users and editors it looks like the field is empty, because the content is hidden. This results in users filling in this field over and over again, thinking that the entered value was not saved.

**To Reproduce**
1. Add a Field to the submission process via the `submission-forms.xml`
2. Add the associated metadata field to metadata.hide
3. Start a new Submission / Submit a new Item as a regular user without administrative rights
4. Enter a value in the new Field
5. Save the Item / Save for later. 
6. The entered value will no longer be visible

If you repeat this process with admin rights, the entered value does not disappear.

**Expected behavior**
Metadata.hide should hide the fields only after publishing and not already in the submission or workflow process, as this causes some confusion

**Comment by uofmsean** [comment]
This is most likely the same issue I reported in DSpace/dspace-angular#1997

**Comment by tdonohue** [comment]
Yes, this is a duplicate of https://github.com/DSpace/dspace-angular/issues/1997 Closing this duplicate ticket as that other ticket is already prioritized on our board.
